RoaringBitmap/src/main/java/org/roaringbitmap/longlong/HighLowContainer.java

```diff
@@ -0,0 +1,166 @@
+   */
+  public LeafNodeIterator highKeyLeafNodeIterator(boolean reverse) {
+    return art.leafNodeIterator(reverse, containers);
+  }
+
+  /**
+   * remove the key and the corresponding container value. Note:While you iterate the art using the
+   * KeyIterator or LeafNodeIterator calling the remove method will make the iterate go to wrong.
+   * The reason is the iterator using the stack to keep the node path,the remove method will cause
+   * the art node adaptive to another one,then make the iterate stack hold a stale node. So
+   * recommend you call the iterator's remove method.
+   */
+  public void remove(byte[] highPart) {
+    long containerIdx = art.remove(highPart);
+    containers.remove(containerIdx);
+  }
+
+  public void replaceContainer(long containerIdx, Container container) {
+    containers.replace(containerIdx, container);
+  }
+
+  public boolean isEmpty() {
+    return art.isEmpty();
+  }
+
+  public void serialize(ByteBuffer byteBuffer) throws IOException {
+    //TODO:
+  }
+
+  public void deserialize(ByteBuffer byteBuffer) throws IOException {
+    //TODO:
+  }
+
+  public long serializedSizeInBytes() {
+    long totalSize = 1l;
+    if (art.isEmpty()) {
+      return totalSize;
+    }
+    totalSize += art.serializeSizeInBytes();
+    totalSize += containers.serializedSizeInBytes();
+    return totalSize;
+  }
+
+  public void serialize(DataOutput dataOutput) throws IOException {
+    if (art.isEmpty()) {
+      dataOutput.writeByte(EMPTY_TAG);
+      return;
+    } else {
+      dataOutput.writeByte(NOT_EMPTY_TAG);
+    }
+    art.serializeArt(dataOutput);
+    containers.serialize(dataOutput);
+  }
+
+  public void deserialize(DataInput dataInput) throws IOException {
+    clear();
+    byte emptyTag = dataInput.readByte();
+    if (emptyTag == EMPTY_TAG) {
+      return;
+    }
+    art.deserializeArt(dataInput);
+    containers.deserialize(dataInput);
+  }
+
+  public void clear() {
+    art = new Art();
+    containers = new Containers();
+  }
+
+  @Override
+  public int hashCode() {
+    int hashCode = 0;
+    KeyIterator keyIterator = highKeyIterator();
+    while (keyIterator.hasNext()) {
+      byte[] key = keyIterator.next();
+      int result = 1;
+      for (byte element : key) {
+        result = 31 * result + element;
+      }
+      long containerIdx = keyIterator.currentContainerIdx();
+      Container container = containers.getContainer(containerIdx);
+      hashCode = 31 * hashCode + result + container.hashCode();
+    }
+    return hashCode;
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (object instanceof HighLowContainer) {
+      HighLowContainer otherHighLowContainer = (HighLowContainer) object;
+      if (this.art.getKeySize() != otherHighLowContainer.art.getKeySize()) {
+        return false;
+      }
+      KeyIterator thisKeyIte = this.highKeyIterator();
+      while (thisKeyIte.hasNext()) {
+        byte[] thisHigh = thisKeyIte.next();
```

Comment:
Could this return a `long` with the bytes in the appropriate positions instead?

Reply:
no additional memory allocation ,not seen any benefit